### ISSUE TITLE
Replay banner does not display in Firefox

### ISSUE THREAD

**Issue by machawk1** [oduwsdl/ipwb#433]
We use [Reconstructive](https://github.com/oduwsdl/reconstructive) for our replay banner (ipwb v0.2018.07.10.1504, 3716d124542c33325d3e188cd35de134904ae11a). It uses Custom Elements for display. Per https://github.com/oduwsdl/MementoEmbed/issues/48#issuecomment-403966880, Custom Elements do not work in Firefox unless a flag is enabled. We can't expect users to do enable this flag for fundamental ipwb functionality. It will work eventually when Firefox supports it without a flag but it remains that there is no graceful degradation for browsers that may never support it.

Chrome 67
![screen shot 2018-07-10 at 5 24 09 pm](https://user-images.githubusercontent.com/2514780/42538591-57a8b37c-8466-11e8-9fc6-eb600c72f62a.png)

Firefox 61
![screen shot 2018-07-10 at 5 24 08 pm](https://user-images.githubusercontent.com/2514780/42538586-531fe578-8466-11e8-9772-d2fb25a0890b.png)

Lets fallback better and allow the replay banner to still be displayed when the browser does not support custom elements.

/cc @ibnesayeed @shawnmjones @phonedude

**Comment by machawk1** [comment]
Have you tested Reconstructive with a polyfill, @ibnesayeed?

**Comment by ibnesayeed** [comment]
Banner is not an essential piece of a memento. While having it is useful, but it is more like a value added feature that we can wait for browsers to catch up (and implement *Custom Elements*). If we really want to make it support Firefox and Edge today, we can add already available polyfills. On the other hand, a more critical piece of functionality that is missing is the ability to rewrite on server-side when we detect that the client does not support *Service Worker*.

**Comment by machawk1** [comment]
> If we really want to make it support Firefox and Edge today, we can add already available polyfills.

Lets do this.

> we detect that the client does not support Service Worker

[SWs](https://caniuse.com/#search=service%20worker) seem fairly well supported but a check and fallback of displaying the information would be useful and more accessible. With various sites' usage of currency miners in Web Workers, I can imagine a scenario where a client has Workers disabled despite their client supporting the capability.

**Comment by ibnesayeed** [comment]
> Have you tested Reconstructive with a polyfill, @ibnesayeed?

No, I did not, because I was not desperately looking for it to work everywhere. The logic and proof of concept was more important initially than wider support. However, I knew that polyfills for this existed for last few years. [Web Component](https://www.webcomponents.org/)'s official site has a happy browser support matrix while [CanIUse?](https://caniuse.com/#feat=custom-elementsv1) suggests about 80% browser market-share already supports it natively.

**Comment by ibnesayeed** [comment]
> Lets do this.

The official polyfill can be found at https://github.com/webcomponents/custom-elements.

**Comment by ibnesayeed** [comment]
With the public release of Firefox 63 last month, Reconstructive banner now supports it natively. @machawk1 should this issue be closed now?

![ipwb-reconstructive-banner-firefox-63](https://user-images.githubusercontent.com/65147/48013056-79342200-e0f1-11e8-83f7-e8f6df62dd30.png)
